packages/plugin/src/classes/helpers/classes.js

```diff
@@ -212,6 +212,40 @@ export function convertClassToUI5Extend(
 
       if (!member.value) continue; // remove all other un-initialized static class props (typescript)
 
+      // Now handle the case where an extended controller extension - not just e.g. "Routing", but
+      // "Routing.override({...})" - is assigned to the class property. To make this work in TypeScript code
+      // (Routing.override({...}) returns a class, but the "routing" member property needs to be typed as an instance),
+      // we require the following notation, using a dummy function ControllerExtension.use() that does actually
+      // not exist at runtime, but does the required class-to-instance conversion from TypeScript perspective:
+      //    routing: ControllerExtension.use(Routing.override({...}))
+      // In this case, the code in the "if" clause above was not executed, regardless of the presence of the
+      // @transformControllerExtension marker, because it is not a type assignment. In the resulting code, the
+      // "ControllerExtension.use(...)" part should be removed and the content of the brackets should be assigned
+      // directly to the member property.
+      if (t.isCallExpression(member.value)) {
+        const callee = member.value.callee;
+        if (
+          t.isMemberExpression(callee) &&
+          t.isIdentifier(callee.object) &&
+          t.isIdentifier(callee.property) &&
+          callee.property.name === "use" // we are looking for "ControllerExtension.use(...)"
+        ) {
+          const importDeclaration = getImportDeclaration(
+            memberPath.hub.file.opts.filename,
```

Comment:
Can it be that something is undefined here?

Reply:
I believe not. These are Babel structures that always exist for a memberPath. I could not create an example that breaks it. Actually no code example at all that makes the code crash.
But for more robustness against Babel changes and unexpected code I will add some checks.